[ANGLE] Update ANGLE to 2024-04-26 (8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17)

djg opened the pull request:
#### c065bad621ef6315cd418a41b67bf1427445d9c3
<pre>
[ANGLE] Update ANGLE to 2024-04-26 (8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17)
<a href="https://bugs.webkit.org/show_bug.cgi?id=273381">https://bugs.webkit.org/show_bug.cgi?id=273381</a>
<a href="https://rdar.apple.com/127211299">rdar://127211299</a>

Reviewed by Kimmo Kinnunen

Contains upstream commits:
git log --oneline 6cfb593174d2f9d432386f4ea1fb587c9c395593..8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17 --pretty=%h %s
8332e5b10e Use python3 instead of python for bootstrap
b74de55781 Vulkan: Add a feature related to shader subsets
d9a008755c Metal: Override internal format of images
89caa0e1d9 Cleanup: replace DirtyObjectType check with constexpr generator
5e37ae35f6 Roll vulkan-deps from a4bf9c4e9ffd to 736920240f0b (2 revisions)
884dc38077 EGL: Make eglGetCurrent(Context|Surface|Display) lockless.
bea8d9dd8f Roll Chromium from ad99ecc02953 to 0129806a25c3 (372 revisions)
c62c0c2500 CL/VK: Implement initial enqueueRead/WriteBuffer

Canonical link: <a href="https://commits.webkit.org/278146@main">https://commits.webkit.org/278146@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f9ef5e5c43ca5512def7ad37555f9d326254d245

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28785 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52541 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/52742 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/176 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34803 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26405 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40410 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51603 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26324 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42647 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21529 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23784 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43831 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7872 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45698 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44336 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54254 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24586 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20765 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47782 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25857 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/42826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46798 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10893 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26697 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25580 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/ThirdParty/ANGLE/src/libANGLE/State.h:
```diff
@@ -1462,13 +1462,40 @@ class State : angle::NonCopyable
-        &State::syncTextures,        &State::syncImages,
-        &State::syncSamplers,        &State::syncProgramPipelineObject};
+    static constexpr std::array<DirtyObjectHandler, state::DIRTY_OBJECT_MAX> kDirtyObjectHandlers =
+        []() {
```
To get GTK pass, we'd need to ifdef this out or remove for the time being.

kkinnunen-apple reviewed (approved):
Please address the linux failure by removing the static assert or something..

djg committed:
[ANGLE] Update ANGLE to 2024-04-26 (8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17)
https://bugs.webkit.org/show_bug.cgi?id=273381
rdar://127211299

Reviewed by Kimmo Kinnunen

Contains upstream commits:
git log --oneline 6cfb593174d2f9d432386f4ea1fb587c9c395593..8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17 --pretty=%h %s
8332e5b10e Use python3 instead of python for bootstrap
b74de55781 Vulkan: Add a feature related to shader subsets
d9a008755c Metal: Override internal format of images
89caa0e1d9 Cleanup: replace DirtyObjectType check with constexpr generator
5e37ae35f6 Roll vulkan-deps from a4bf9c4e9ffd to 736920240f0b (2 revisions)
884dc38077 EGL: Make eglGetCurrent(Context|Surface|Display) lockless.
bea8d9dd8f Roll Chromium from ad99ecc02953 to 0129806a25c3 (372 revisions)
c62c0c2500 CL/VK: Implement initial enqueueRead/WriteBuffer

Canonical link: https://commits.webkit.org/278146@main
